### PR TITLE
fix: activity id not using overridden id

### PR DESCRIPTION
**Pull request by EmosewaMC**
Resolves an issue where starting the Nimbus Station race sends you to Gnarled Forest race.

### CODE REVIEW

File: dGame/dComponents/ActivityComponent.cpp [hunk 1]
```diff
@@ -30,23 +30,11 @@
...
-			m_ActivityInfo.minTeams = 1;
-		}
-		if (m_ActivityInfo.instanceMapID == -1) {
-			const auto& transferOverride = parent->GetVarAsString(u"transferZoneID");
-			if (!transferOverride.empty()) {
-				GeneralUtils::TryParse(transferOverride, m_ActivityInfo.instanceMapID);
-			}
-		}
+	m_ActivityID = activityID;
+	LoadActivityData(activityID);
```
Comment (Xiphoseer) [inline dGame/dComponents/ActivityComponent.cpp:34]: Is there a reason not to do this only once after the `if`?

Comment (EmosewaMC) [reply dGame/dComponents/ActivityComponent.cpp:34]: This is precisely what the client does functionally
Use the component id as the default activity id and load its data from the database
if activityID is specified and if that column exists in the activities table, update the activity info with that data.
This is what actually resolves the NT parrot foot race issue and why DLU inserted a activity 49 into the database in the first place.  It used all of the data from 39 by default, tried to load 49, found nothing, and used 49 for the leaderboards and all the data for 39.  Will add a comment.

File: dGame/dComponents/ActivityComponent.cpp [hunk 2]
```diff
@@ -74,6 +68,24 @@ ActivityComponent::ActivityComponent(Entity* parent, int32_t activityID) : Compo
 		}
 	}
 }
+void ActivityComponent::LoadActivityData(const int32_t activityId) {
+	CDActivitiesTable* activitiesTable = CDClientManager::Instance().GetTable<CDActivitiesTable>();
+	std::vector<CDActivities> activities = activitiesTable->Query([activityId](CDActivities entry) {return (entry.ActivityID == activityId); });
+
+	for (CDActivities activity : activities) {
+		m_ActivityInfo = activity;
+		if (static_cast<Leaderboard::Type>(activity.leaderboardType) == Leaderboard::Type::Racing && Game::config->GetValue("solo_racing") == "1") {
```
Comment (Xiphoseer) [inline dGame/dComponents/ActivityComponent.cpp:77]: Actually, you may want to pull this GetValue out of the loop

**Review by Xiphoseer — DISMISSED** [review]

File: dGame/dComponents/ActivityComponent.cpp (hunk 2 above)
Comment (EmosewaMC) [reply dGame/dComponents/ActivityComponent.cpp:77]: Any particular reason?

**Review by aronwk-aaron — DISMISSED** [review]

File: dGame/dComponents/ActivityComponent.cpp (hunk 2 above)
Comment (Xiphoseer) [reply dGame/dComponents/ActivityComponent.cpp:77]: The env reading code that might potentially allocate and it's very unlikely to change between iterations

Comment (EmosewaMC) [reply dGame/dComponents/ActivityComponent.cpp:77]: how are you supposed to know whether to use solo racing then?

Comment (Xiphoseer) [reply dGame/dComponents/ActivityComponent.cpp:77]:
```
bool soloRacing = Game::config->GetValue("solo_racing") == "1";
for (...) {
  if (... && soloRacing) ...
  ...
}
```
is what I mean.

Comment (EmosewaMC) [reply dGame/dComponents/ActivityComponent.cpp:77]: Ah that was not clear.  Will change this then.

**Review by Xiphoseer — APPROVED** [review]

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**